packages/js/product-editor/src/components/schedule-sale-pricing-block/edit.tsx

```diff
@@ -0,0 +1,149 @@
+/**
+ * External dependencies
+ */
+import { DateTimePickerControl } from '@woocommerce/components';
+import { OPTIONS_STORE_NAME } from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+import { useBlockProps } from '@wordpress/block-editor';
+import { BlockEditProps } from '@wordpress/blocks';
+import { ToggleControl } from '@wordpress/components';
+import { useEntityProp } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import { createElement, useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import moment from 'moment';
+
+/**
+ * Internal dependencies
+ */
+import { ScheduleSalePricingBlockAttributes } from './types';
+import { useValidation } from '../../hooks/use-validation';
+
+export function Edit( {}: BlockEditProps< ScheduleSalePricingBlockAttributes > ) {
+	const blockProps = useBlockProps( {
+		className: 'wp-block-woocommerce-product-schedule-sale-pricing',
+	} );
+
+	const dateTimeFormat = useSelect( ( select ) => {
+		const { getOption } = select( OPTIONS_STORE_NAME );
+		return (
+			getOption< string | null >( 'links_updated_date_format' ) ||
+			'F j, Y g:i a'
+		);
+	} );
+
+	const [ showSaleSchedule, setShowSaleSchedule ] = useState( false );
```

Comment:
Very minor:

```suggestion
	const [ showScheduleSale, setShowScheduleSale ] = useState( false );
```

Since `schedule` at the end might imply the noun of "schedule" (like a timetable) instead of the verb.

Reply:
Done -> https://github.com/woocommerce/woocommerce/pull/37567/commits/ffe64739b7e1aafcedd477eb7d7fbeefaae05999